src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1840,82 +1840,68 @@ private void InferTypeFrom(VariableExpressionAst variableExpressionAst, List<PST
                 (SpecialVariables.IsUnderbar(astVariablePath.UserPath)
                  || astVariablePath.UserPath.EqualsOrdinalIgnoreCase(SpecialVariables.PSItem)))
             {
-                // $_ is special, see if we're used in a script block in some pipeline.
-                while (parent != null)
+                bool hasSeenScriptBlock = false;
+                while (parent is not null)
                 {
-                    if (parent is ScriptBlockExpressionAst || parent is CatchClauseAst)
+                    if (parent is CatchClauseAst)
                     {
                         break;
                     }
-
-                    parent = parent.Parent;
-                }
-
-                if (parent != null)
-                {
-                    if (parent.Parent is CommandExpressionAst && parent.Parent.Parent is PipelineAst)
+                    else if (parent is ScriptBlockExpressionAst)
                     {
-                        // Script block in a hash table, could be something like:
-                        //     dir | ft @{ Expression = { $_ } }
-
-                        if (parent.Parent.Parent.Parent is HashtableAst)
+                        hasSeenScriptBlock = true;
+                    }
+                    else if (hasSeenScriptBlock)
+                    {
+                        if (parent is InvokeMemberExpressionAst invokeMember)
```

Comment:
Can you please add a comment here explaining what this scenario is (with an example)?